Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,247 @@
+
+                if (!isLow) break;
+            }
+
+            // It's possible case when the bar forms both low and high points at the same time
+            if (isHigh && isLow)
+            {
+                var pointHigh = new PivotPoint(PivotPointType.High, middlePoint1.High, middlePoint1.EndTime);
+                _pivotPoints.Add(pointHigh);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(pointHigh));
+
+                var pointLow = new PivotPoint(PivotPointType.Low, middlePoint2.Low, middlePoint2.EndTime);
+                _pivotPoints.Add(pointLow);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(pointLow));
+
+                return 2m;
+            }
+
+            if (isHigh)
+            {
+                var point = new PivotPoint(PivotPointType.High, middlePoint1.High, middlePoint1.EndTime);
+                _pivotPoints.Add(point);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(point));
+
+                return 1m;
+            }
+
+            if (isLow)
+            {
+                var point = new PivotPoint(PivotPointType.Low, middlePoint2.Low, middlePoint2.EndTime);
+                _pivotPoints.Add(point);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(point));
+
+                return -1m;
+            }
+
+            return 0m;
+        }
+
+        /// <summary>
+        /// Get current high pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of high pivot points.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetHighPivotPointsArray()
+        {
+            return _pivotPoints?.Where(p => p.PivotPointType == PivotPointType.High).ToArray();
+        }
+
+        /// <summary>
+        /// Get current low pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of low pivot points.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetLowPivotPointsArray()
+        {
+            return _pivotPoints?.Where(p => p.PivotPointType == PivotPointType.Low).ToArray();
+        }
+
+        /// <summary>
+        /// Get all pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of low and high pivot points. Ordered by time in descending order.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetAllPivotPointsArray()
+        {
+            // Get all pivot points within rolling wind. collection as an array
+            return _pivotPoints?.Take(_pivotPoints.Count).ToArray();
```

Comment:
`_pivotPoints` can't be null? Don't think we need the null check

Reply:
I think null conditional operator takes care of it here